modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/test/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/util/DDMFormInstanceSubmissionLimitStatusUtilTest.java

```diff
@@ -0,0 +1,163 @@
+					_mockDDMFormInstance(_mockDDMFormInstanceSettings(true))));
+	}
+
+	@Test
+	public void testIsSubmissionLimitReached() throws Exception {
+		Assert.assertTrue(
+			DDMFormInstanceSubmissionLimitStatusUtil.isSubmissionLimitReached(
+				_mockDDMFormInstance(_mockDDMFormInstanceSettings(true)),
+				_mockDDMFormInstanceRecordVersionLocalService(
+					Collections.singletonList(
+						mock(DDMFormInstanceRecordVersion.class))),
+				mock(User.class)));
+	}
+
+	@Test
+	public void testIsSubmissionLimitReachedWithDefaultUser() throws Exception {
+		User user = mock(User.class);
+
+		when(
+			user.isDefaultUser()
+		).thenReturn(
+			true
+		);
+
+		Assert.assertFalse(
+			DDMFormInstanceSubmissionLimitStatusUtil.isSubmissionLimitReached(
+				_mockDDMFormInstance(_mockDDMFormInstanceSettings(true)),
+				_mockDDMFormInstanceRecordVersionLocalService(
+					Collections.singletonList(
+						mock(DDMFormInstanceRecordVersion.class))),
+				user));
+	}
+
+	@Test
+	public void testIsSubmissionLimitReachedWithNoRecord() throws Exception {
+		Assert.assertFalse(
+			DDMFormInstanceSubmissionLimitStatusUtil.isSubmissionLimitReached(
+				_mockDDMFormInstance(_mockDDMFormInstanceSettings(true)),
+				_mockDDMFormInstanceRecordVersionLocalService(
+					Collections.emptyList()),
+				mock(User.class)));
+	}
+
+	@Test
+	public void testIsSubmissionLimitReachedWithStatusDraft() throws Exception {
+		DDMFormInstanceRecordVersion ddmFormInstanceRecordVersion = mock(
+			DDMFormInstanceRecordVersion.class);
+
+		when(
+			ddmFormInstanceRecordVersion.getStatus()
+		).thenReturn(
+			WorkflowConstants.STATUS_DRAFT
+		);
+
+		Assert.assertFalse(
+			DDMFormInstanceSubmissionLimitStatusUtil.isSubmissionLimitReached(
+				_mockDDMFormInstance(_mockDDMFormInstanceSettings(true)),
+				_mockDDMFormInstanceRecordVersionLocalService(
+					Collections.singletonList(ddmFormInstanceRecordVersion)),
+				mock(User.class)));
+	}
+
+	private DDMFormInstance _mockDDMFormInstance(
+			DDMFormInstanceSettings ddmFormInstanceSettings)
+		throws Exception {
+
+		DDMFormInstance ddmFormInstance = mock(DDMFormInstance.class);
+
+		when(
+			ddmFormInstance.getSettingsModel()
+		).thenReturn(
+			ddmFormInstanceSettings
+		);
+
+		return ddmFormInstance;
+	}
+
+	private DDMFormInstanceRecordVersionLocalService
+		_mockDDMFormInstanceRecordVersionLocalService(
+			List<DDMFormInstanceRecordVersion> ddmFormInstanceRecordVersions) {
+
+		DDMFormInstanceRecordVersionLocalService
+			ddmFormInstanceRecordVersionLocalService = mock(
+				DDMFormInstanceRecordVersionLocalService.class);
+
+		when(
+			ddmFormInstanceRecordVersionLocalService.
+				getFormInstanceRecordVersions(
+					Matchers.anyLong(), Matchers.anyLong())
+		).thenReturn(
+			ddmFormInstanceRecordVersions
+		);
+
+		return ddmFormInstanceRecordVersionLocalService;
+	}
+
+	private DDMFormInstanceSettings _mockDDMFormInstanceSettings(
+			boolean limitToOneSubmissionPerUser)
+		throws Exception {
```

Comment:
I don't think we need to throw an exception here.

Reply:
Done!